### ISSUE TITLE
The carpentries.org content redesign

### ISSUE THREAD

**Issue by kariljordan** [carpentries/strategic-plan#27]
Carpentries Team dedicates time to read, research and learn about web accessibility as a necessary prerequisite to signing up for a design thinking workshop as a team in Phase 2 and the start of website redesign work in phase 3.

The three phases are: 
- [x] phase 1: deep dive into web accessibility (Q4 2020)
- [x] phase 2: design thinking workshop with web accessibility in mind (Q1 and Q2 2021)
- [x] phase 3: website redesign (Q2 2021 onwards)

**this section was updated to reflect a necessary shift in the focus of each of the three phases of this project**

**Comment by serahkiburu** [comment]
On May 3 2021, we kicked-off work with a designer with the following objectives:

The new Carpentries website should:
- **Prioritise discoverability**. The new website should make it as easy as possible for community members to find the information they need and act on it to accomplish their goals with little to no extra help from others.
- **Reflect who we are and what we do**. Users of the website outside of our community should understand our mission, vision, and values from the landing page. They should be able to easily identify how they could get involved with us based on their background.
- **Conform to web accessibility standards**. The new website and handbook will need to comply with the Web Content Accessibility Guidelines (WCAG) 2.1, so that users having disabilities that affect how they visually or physically interact with the website and handbook are not impeded.
- **Accommodate resource translations** with ease of navigation between content in different languages on each page. We are a global community and we want to be able to provide (at least some of) the content of our site in multiple languages. 
- **Offer the same usability regardless of the device it is accessed from**. Many of our users have limited Internet access and only have access to older hardware. Everyone should have a similar experience of the site regardless of the device or quality of their internet connection. 
- **The styling elements of the website and handbook (CSS, JS) should be lightweight and broadly compatible**. Because our community is globally distributed, many of our users have limited Internet access and only have access to older hardware. The website and handbook should therefore use little bandwidth, have styling primarily through CSS to be minimally usable with JavaScript disabled, and compatible with older web browsers.

Note that the revision of the lesson program websites is out-of-scope for this project.

We expect that design work will take 10 - 12 weeks, and hope to work with a frontend developer to actualise the approved designs for 9 weeks after the design work is complete. We hope to launch a new website in late 2021 or early 2022.

**Comment by acrall** [comment]
Is this currently out of the hands of the CDT until the web designer has finished her work?

**Comment by kariljordan** [comment]
@kmomar is the CDT representative who is helping with the redesign. Once the web designer is finished, we will have a developer code the website, and that will be the end of phase 3 @acrall.

**Comment by fmichonneau** [comment]
The web designer will deliver the final product of the work next week. After that, we will work with a web developer to create a template that we can use for our main website, and that will be a starting point to harmonize our visual identity across all our sites.

**Comment by kariljordan** [comment]
The Carpentries is keen to improve accessibility across our interactions and resources. We approached this in three ways. Firstly, we engaged the community in a [conversation](https://github.com/carpentries/conversations/issues/26) about equitable access to and availability of platforms and tools used for collaboration in online communities. We collated feedback and populated accessibility related issues on our [Help Wanted](https://carpentries.org/help-wanted-issues/) page. A few examples of issues and PRs submitted include:
- expanding word contractions across our instructor training, handbook and website resources as[ using contractions makes content inaccessible for some](https://medium.com/@joanne.schofield/using-contractions-could-be-making-your-writing-inaccessible-4034bb3cd76a);
- making links more descriptive across our resources[ so they are more accessible to people who use screen readers](https://webaccess.berkeley.edu/ask-pecan/click-here);
- improving how links are presented across our resources[ for better readability](https://webaim.org/techniques/hypertext/link_text#appearance); and
- appending alternative descriptions to images across our resources ++

Additionally, the Core Team participated in an [‘Acc-athon’ to add alt text across Carpentries curricula](https://carpentries.org/blog/2021/04/Acc-athon/).

Secondly, we contracted UX/UI designer Oliva DuVal to do a brand refresh comprising the redesign of our organizational and lesson program websites, and our community handbook with the following objectives in mind:

- Prioritise discoverability. The new website should make it as easy as possible for community members to find the information they need and act on it to accomplish their goals with little to no extra help from others.
- Reflect who we are and what we do. Users of the website outside of our community should understand our mission, vision, and values from the landing page. They should be able to easily identify how they could get involved with us based on their background.
- Conform to web accessibility standards. The new website and handbook will need to comply with the [Web Content Accessibility Guidelines (WCAG) 2.1](https://www.w3.org/TR/WCAG21/), so that users having disabilities that affect how they visually or physically interact with the website and handbook are not impeded.
- Accommodate resource translations with ease of navigation between content in different languages on each page. We are a global community and we want to be able to provide (at least some of) the content of our site in multiple languages. 
- Offer the same usability regardless of the device it is accessed from. Many of our users have limited Internet access and only have access to older hardware. Everyone should have a similar experience of the site regardless of the device or quality of their internet connection. 
- The styling elements of the website and handbook (CSS, JS) should be lightweight and broadly compatible. Because our community is globally distributed, many of our users have limited Internet access and only have access to older hardware. The website and handbook should therefore use little bandwidth, have styling primarily through CSS to be minimally usable with JavaScript disabled, and compatible with older web browsers.

Olivia produced a website schema meeting each of these objectives. In doing so, we realized much of the content on our website should be organized into separate handbooks according to various community roles, and that work is being managed by the Community Development Team.

Lastly, we contracted [bytes.co](https://bytes.co/), a web design, digital marketing and development agency specializing in web accessibility. Bytes.co worked in collaboration with their sister company, Accessible Web, to convert the carpentries.org website template schema developed by Olivia DuVal (UX/UI contractor) into static html/css/js web pages. These web pages are currently being converted into full static sites and conform to WCAG 2.1 AA web accessibility standards.

**Comment by acrall** [comment]
Content for the website and handbooks has been developed by each of the teams. The Core Team members assigned to this work are now moving the content into the appropriate GitHub templates.

**Comment by acrall** [comment]
The content developed by teams has been transferred to the appropriate GitHub templates. There is a final round of functionality updates needed before moving into community review and launch. Tagging @Talishask in case she would like to add more

**Comment by froggleston** [comment]
The Website working group has decided not to continue the project with bytes.co and so this will need further discussion and decision making. A full request for quote (RFQ) will be provided by the team to set out our requirements before going to tender. In the interim we will focus on improving the existing website content.

**Comment by acrall** [comment]
The website working group met this week and will begin meeting every two weeks. @OscarSiba, our new Communications Manager, has joined the working group. First task is to finalise the drafted RFQ and share broadly to identify a website developer to complete this project

**Comment by acrall** [comment]
The website working group is interviewing potential contractors next week

**Comment by acrall** [comment]
A contractor has been identified and will start work in the new year. @OscarSiba  and @Talishask have pulled 99% of the website page drafts into a Google document. I will be sharing links to all relevant documentation and provide status update for the working group to continue its work in 2024

**Comment by maneesha** [comment]
Since the last update was at the end of 2023, I am including a summary of all 2024 activity here.  Note also that while the title of this issue applies just to carpentries.org, this update refers to that site as well as the three lesson program sites and the handbook.  

We hired The New Dynamic to work with us at the end of 2023.  After hiring The New Dynamic to build a Hugo theme for us, we spent the first half of 2024 working with them on this process.  The intent was to have one theme used for all four websites (The Carpentries and its three lesson programs) so all four websites would have the same look and feel and user experience.  This process began by sharing our designs with them, testing the theme they provided, sharing feedback with them, and continuing this process until all deliverables were met.

As we developed the theme, we would populate it with real content as provided by Core Team members.  As of this status update (2024-11-14), we have been able to implement the theme and content and will be completing deployment by 2024-11-18.

At the same time, we have been working on revamping our Handbook to update content, remove outdated content, and update the general theme and organization to make resources easier to access.  The development of the new handbook followed a similar pattern of testing and user feedback.  The new handbook will be deployed 2024-12-09.

Both of these launches are intended to be "soft launches." We have many outstanding issues in the relevant GH repos tagged "later" that we will come back to in 2025. 

Edit to add note - @OscarSiba please feel free to add any other comments you would like to include.

**Comment by maneesha** [comment]
The websites for The Carpentries and the three lesson programs and the Carpentries Handbook were launched in 2024 Q4.  Over the course of the year in 2025, we implemented systems to streamline maintenance of all these resources, including a GitHub project that lets us track all open issues by individual and team assignment, priority, and timeline.  The website management team (Maneesha & Oscar) meets regularly to review outstanding issues and plan for ongoing maintenance and improvements to these resources.  

@OscarSiba please feel free to add any other comments you would like to include.
